src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs

```diff
@@ -2508,23 +2508,16 @@ protected internal override Expression VisitRuntimeVariables(RuntimeVariablesExp
                 var indexes = new int[count];
                 for (int i = 0; i < count; i++)
                 {
-                    LocalVariable var;
-                    if (_variables.TryGetValue(node.Variables[i], out var))
+                    IStrongBox box = GetBox(node.Variables[i]);
```

Comment:
With these changes, the code of `VisitBlock` is identical between compiler and interpreter.